flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java

```diff
@@ -190,21 +193,21 @@ public boolean equals(Object o) {
             return timestamp == that.timestamp
                     && Objects.equals(exception, that.exception)
                     && Objects.equals(task, that.task)
-                    && Objects.equals(location, that.location)
+                    && Objects.equals(endpoint, that.endpoint)
                     && Objects.equals(taskManagerId, that.taskManagerId);
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(timestamp, exception, task, location, taskManagerId);
+            return Objects.hash(timestamp, exception, task, endpoint, taskManagerId);
```

Comment:
This means that if we get an old object with location in it , then this will hash to a different value to the new object with the endpoint. Is this what we want? Or do we want them to match if the hostname and port are the same?